pkg/operator/profilecalculator.go

```diff
@@ -167,55 +169,100 @@ func (pc *ProfileCalculator) calculateProfile(nodeName string) (string, map[stri
 		return "", nil, operand, fmt.Errorf("failed to list Tuned: %v", err)
 	}
 
-	for _, recommend := range TunedRecommend(tunedList) {
-		var (
-			pools []*mcfgv1.MachineConfigPool
-			node  *corev1.Node
-		)
+	recommendAll := TunedRecommend(tunedList)
+	r := func(nodeName string, iStart int) (int, string, map[string]string, tunedv1.OperandConfig, error) {
```

Comment:
let's use a more descriptive name - and in any case let's try hard to avoid single-letter identifiers except for some loop variables

Reply:
The lambda is actually rather long, I think giving it a proper top level name might be more readable.

Reply:
FWIR, it would require bigger code changes to remove the lambda and move it outside of the function, but leaving this to Ronny. My suggestion is to just rename the `r()` to something like `recommendProfile()` and perhaps document the parameters and values returned.

Reply:
@MarSik I was thinking of moving the lambda to the top level but it does require passing more objects as parameters and a little more effort. Note this would be required to happen for both `calculateProfile` and `calculateProfileHyperShift` implementations (As their `r` function is a bit different). If you think this would be more readable to create these two outer functions I will make the changes